hoot-core/src/main/cpp/hoot/core/ops/MapCropper.cpp

```diff
@@ -302,29 +272,23 @@ void MapCropper::apply(OsmMapPtr& map)
     const Envelope& wayEnv = *(ls->getEnvelopeInternal());
     LOG_VART(wayEnv);
 
-    if (_inclusionCrit)
-    {
-      LOG_VART(_inclusionCrit->isSatisfied(w));
-    }
-    if (_inclusionCrit && _inclusionCrit->isSatisfied(w))
-    {
-      // keep the way
-      LOG_TRACE("Keeping explicitly included way: " << w->getElementId() << "...");
-      _explicitlyIncludedWayIds.insert(w->getId());
-      _numWaysInBounds++;
-    }
-    // Have found that if we have _envelopeG, using it for inside/outside way comparison is more
-    // accurate than using the envelope. Its possible that we could eventually use that method
-    // exclusively and get rid of storing _envelope.
-    else if ((_envelopeG && _isWhollyOutside(*ls)) || _isWhollyOutside(wayEnv))
+    // It seems very unnecessary to check against both the way's linestring geometry and its
+    // envelope here, however, this is how this was originally written after the option to check
+    // against a geometry was added (the class originally only checked against envelopes). Several
+    // test failures occur if you just try to check one or the other (checking against the
+    // linestring geometry seems to make more sense...but maybe not...). Checking both could
+    // contribute to crop performance issues. Opened #4359 to look further into it.
+    if (_isWhollyOutside(wayEnv) || _isWhollyOutside(*ls))
```

Comment:
This is a little strange to me with the multiple checks. Try as I could, I couldn't prevent test output havoc by using one or the other. Opened an issue for it.